web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/Plugin/Field/FieldFormatter/ViewsBasicDefaultFormatter.php

```diff
@@ -99,26 +113,39 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
 
     $elements = [];
     foreach ($items as $delta => $item) {
-      $paramsForRender = [
-        'types' => [],
-        'view_mode' => '',
-      ];
-      $paramsDecoded = json_decode($item->params, TRUE);
 
-      // Gets the entity labels.
-      foreach ($paramsDecoded['filters']['types'] as $type) {
-        $entityLabel = $this->viewsBasicManager->getEntityLabel($type);
-        array_push($paramsForRender['types'], $entityLabel);
+      // Prevents views recursion.
+      static $running;
+      if ($running) {
+        return $elements;
       }
+      $running = TRUE;
+
+      // Set up the view and initial decoded parameters.
+      $view = Views::getView('views_basic_scaffold');
+      $view->setDisplay('block_1');
+      $paramsDecoded = json_decode($item->getValue()['params'], TRUE);
+
+      // Overrides filters using our custom views filter - ViewsBasicFilter.
+      $filters = $view->display_handler->getOption('filters');
+      $filters['views_basic_filter']['value'] = $paramsDecoded;
```

Comment:
For now this is exactly what we need. But I think in the next iteration, we should move some of this to the manager service. I don't think the views plugin should worry about decoding the params and then finding the right value. That's all business logic that can exist in the single management class. 

Instead I would have a method for getFilter() or getFitlers() or getFiltersbyType() or whatever we may need. The management class returns a string. It can even do validation and default values if needed. It's likely other views plugins will need similar functions, so might as well centralize them.

No change at the moment. But I think we will add some additional filters next sprint and that will be the time to experiment a little.